app/src/main/java/org/simple/clinic/patient/PatientRepository.kt

```diff
@@ -36,19 +37,30 @@ class PatientRepository @Inject constructor(
 
   private var ongoingPatientEntry: OngoingPatientEntry = OngoingPatientEntry()
 
-  fun searchPatientsAndPhoneNumbers(query: String?): Observable<List<PatientSearchResult>> {
+  fun searchPatientsAndPhoneNumbers(query: String?, includeFuzzyNameSearch: Boolean = true): Observable<List<PatientSearchResult>> {
     if (query.isNullOrEmpty()) {
       return database.patientSearchDao()
           .recentlyUpdated100Records()
           .toObservable()
     }
 
-    return database.patientSearchDao()
-        .search(nameToSearchableForm(query!!))
-        .toObservable()
+    val actualQuery = nameToSearchableForm(query!!)
+
+    return if (actualQuery.all { it.isDigit() } || includeFuzzyNameSearch.not()) {
+      database.patientSearchDao()
+          .search(actualQuery)
+          .toObservable()
+    } else {
+      database.patientSearchDao()
+          .search(actualQuery)
+          .zipWith(database.fuzzyPatientSearchDao().searchForPatientsWithNameLike(actualQuery).toFlowable()) { searchResults, fuzzySearchResults ->
+            (fuzzySearchResults + searchResults).distinctBy { it.uuid }
+          }
+          .toObservable()
+    }
```

Comment:
Can you extract the argument to `zipWith()` as a separate value? There's a lot happening in a single line there. I tried moving things around on my machine:

```kotlin
val fuzzySearch = database.fuzzyPatientSearchDao()
    .searchForPatientsWithNameLike(actualQuery)
    .toObservable()

database.patientSearchDao()
    .search(actualQuery)
    .toObservable()
    .zipWith(fuzzySearch)
    .map { (results, fuzzyResults) -> (fuzzyResults + results).distinctBy { it.uuid } }
```

Reply:
Yes, this looks a lot nicer. I'll do it this way.